src/fu-util.c

```diff
@@ -302,54 +181,6 @@ fu_util_prompt_for_device (FuUtilPrivate *priv, GError **error)
 	return g_object_ref (dev);
 }
 
-static gboolean
-fu_util_setup_networking (FuUtilPrivate *priv, GError **error)
-{
-	const gchar *http_proxy;
-	g_autofree gchar *user_agent = NULL;
-
-	/* already done */
-	if (priv->soup_session != NULL)
-		return TRUE;
-
-	/* create the soup session */
-	user_agent = fwupd_build_user_agent (PACKAGE_NAME, PACKAGE_VERSION);
-	priv->soup_session = soup_session_new_with_options (SOUP_SESSION_USER_AGENT, user_agent,
-							    SOUP_SESSION_TIMEOUT, 60,
-							    NULL);
-	if (priv->soup_session == NULL) {
-		g_set_error_literal (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
-				     "failed to setup networking");
-		return FALSE;
-	}
-
-	/* set the proxy */
-	http_proxy = g_getenv ("https_proxy");
-	if (http_proxy == NULL)
-		http_proxy = g_getenv ("HTTPS_PROXY");
-	if (http_proxy == NULL)
-		http_proxy = g_getenv ("http_proxy");
-	if (http_proxy == NULL)
-		http_proxy = g_getenv ("HTTP_PROXY");
-	if (http_proxy != NULL) {
-		g_autoptr(SoupURI) proxy_uri = soup_uri_new (http_proxy);
-		if (proxy_uri == NULL) {
-			g_set_error (error,
-				     FWUPD_ERROR,
-				     FWUPD_ERROR_INTERNAL,
-				     "invalid proxy URI: %s", http_proxy);
-			return FALSE;
-		}
-		g_object_set (priv->soup_session, SOUP_SESSION_PROXY_URI, proxy_uri, NULL);
-	}
-
-	/* this disables the double-compression of the firmware.xml.gz file */
-	soup_session_remove_feature_by_type (priv->soup_session, SOUP_TYPE_CONTENT_DECODER);
-	return TRUE;
-}
-
```

Comment:
`fu-tool` intentionally forks out to `wget` or `curl` to be safer in terms of network comms.  I'm not sure there is a reason to move this into the common source file.

Reply:
I'm about to use it for a new CLI -- I don't mind including this commit on that branch if you want. Just seemed nicer on this PR.

Reply:
Oh as long as there is bigger picture stuff for it, that's fine then.